src/fallback/stage1_find_marks.h

```diff
@@ -127,7 +127,7 @@ really_inline error_code scan() {
   }
   *next_structural_index = len;
   next_structural_index++;
-  doc_parser.n_structural_indexes = next_structural_index - doc_parser.structural_indexes.get();
+  doc_parser.n_structural_indexes = static_cast<uint32_t>(next_structural_index - doc_parser.structural_indexes.get());
```

Comment:
Or you can do `static_cast<decltype<doc_parser.n_structural_indexes>>` which has the benefit that you avoid locking yourself into uint32_t. :-)

There may be a more clever way.

Reply:
Yeah, I was thinking about making a typedef for the size to handle this stuff. But I figured we could add it in when we actually toggle uint32_t to size_t; if I just did it where there were warning fixes we'd likely miss some.

Reply:
There is something weird to the C++ culture, where doing "a = b" works all the time even when a and b are different types, and we are totally fine with it... but then, if we are dealing with primitives, we don't trust the magic anymore.